daemon/boardd/queue.hpp

```diff
@@ -2,64 +2,67 @@
 // The MIT License.
 #ifndef _QUEUE_H_
 #define _QUEUE_H_
-#include <queue>
-#include <mutex>
-#include <condition_variable>
+#include<condition_variable>
+#include<mutex>
+#include<queue>
+#include<stdexcept> //runtime_error
+#include<utility> //forward, move
 
-template <class T>
+template<class T>
 class Queue
 {
+	std::queue<T> qu_;
+	std::mutex mu_;
+	std::condition_variable cv_pushed_;
+	std::condition_variable cv_poped_;
+	bool eof_;
+	std::size_t max_;
 public:
-  Queue(size_t max = 0)
-    : eof_(false)
-    , max_(max)
-  {}
-
-  void Push(T elem)
-  {
-    std::unique_lock<std::mutex> lk(mu_);
-    while (true) {
-      if (eof_)
-        throw std::runtime_error("Queue closed");
-      if (max_ && qu_.size() >= max_) {
-        cv_poped_.wait(lk);
-        continue;
-      }
-      qu_.push(elem);
-      cv_pushed_.notify_one();
-      return;
-    }
-  }
-
-  bool Pop(T &elem)
-  {
-    std::unique_lock<std::mutex> lk(mu_);
-    while (qu_.empty()) {
-      if (eof_)
-        return false;
-      cv_pushed_.wait(lk);
-    }
-    elem = qu_.front();
-    qu_.pop();
-    cv_poped_.notify_one();
-    return true;
-  }
-
-  void Close()
-  {
-    std::unique_lock<std::mutex> lk(mu_);
-    eof_ = true;
-    cv_pushed_.notify_all();
-    cv_poped_.notify_all();
-  }
-
-private:
-  std::queue<T> qu_;
-  std::mutex mu_;
-  std::condition_variable cv_pushed_;
-  std::condition_variable cv_poped_;
-  bool eof_;
-  size_t max_;
+	Queue() noexcept
+		:Queue{}{}
+	Queue(std::size_t max) noexcept
+		:eof_{false},max_{max}{}
+	//probably need enable_if and is_convertible
+	template<class T2>
+	void Push(T2 &&elem)
+	{
+		std::unique_lock<std::mutex> lk{mu_};
+		while(true)
+		{
+			if(eof_)
+				throw std::runtime_error{"Queue closed"};
+			if(max_&&qu_.size()>=max_)
+			{
+				cv_poped_.wait(lk); //do you aware spurious wakeup?
+				                    //if not, you may need a semaphore
+				continue;
+			}
+			qu_.emplace(std::forward<T2>(elem));
+			cv_pushed_.notify_one();
+			return;
+		}
+	}
+	bool Pop(T &elem)
+	{
+		std::unique_lock<std::mutex> lk{mu_};
+		while(qu_.empty())
+		{
+			if(eof_)
+				return false;
+			cv_pushed_.wait(lk);  //again, how about spurious wakeup?
+		}
+		elem=std::move(qu_.front());  //why don't you use move?
+		qu_.pop();
+		cv_poped_.notify_one();
+		return true;
+	}
+	void Close()
+	{
+		std::unique_lock<std::mutex> lk{mu_};
+		eof_=true;
+		cv_pushed_.notify_all();
+		cv_poped_.notify_all();
+	}
```

Comment:
Please don't reindent the whole file. It makes changes less obvious, and doesn't play nice with blame either.
Use whatever code style the original file had.